### ISSUE TITLE
The infer speed is much slower than paper said

### ISSUE THREAD

**Issue by JackyChan97** [Thmen/EGVSR#8]
How to upscale to 4K video with 30fps as Fig.7 said?

**Comment by QuantumLiu** [comment]
> How to upscale to 4K video with 30fps as Fig.7 said?

 Hi what your infer speed?

**Comment by JackyChan97** [comment]
I added the code in "codes/models/networks/egvsr_net.py"
```python 
if __name__ == "__main__":
    # from torchsummary import summary

    # img_size = (960, 640)
    # cpu_cuda = 'cuda'
    #
    # device = torch.device(cpu_cuda)
    #
    # model = ResidualBlock()
    # model.eval()
    # model.to(device)
    #
    # print(model)
    # summary(model, (64, *img_size), batch_size=1, device=cpu_cuda)
    from time import time
    model = FRNet(in_nc=3,out_nc=3,nf=64,nb=10).cuda()
    lr = torch.randn([5,3,270,480]).cuda()
    # hr = torch.randn([1,3,270*4,480*4]).cuda()
    t = time()
    with torch.no_grad():
        for _ in range(20):
            model.infer_sequence(lr,torch.device('cuda'))

    torch.cuda.synchronize()
    print(time()-t)
```


And I run the code in computer with i7-10700 and rtx 2060s, the running time was 41.301308393478394

**Comment by ChenyangQiQi** [comment]
# Questions about inference FLOPS and time
Thanks for the great work and clear codes. Following are my issues.
## 1. FPS of my experiment is much lower
I run the profile.sh on 2080Ti but the result FPS 8.2 is much lower than that in Table 4 of the main paper 29.61.
![image](https://user-images.githubusercontent.com/45789244/139401916-aea64ef3-8a36-401a-aecb-8a1e6b00045b.png)

While for tecoGAN, the result is similar
![image](https://user-images.githubusercontent.com/45789244/139402239-676c91b7-4173-4c1e-942a-950649c4c8a8.png)

I compare the code of "models/networks/egvsr_nets.py" and "models/networks/tecogan_nets.py". 
They are basically the same thing. Only the up sampling method is different
I am not sure what is wrong with my profiling

![image](https://user-images.githubusercontent.com/45789244/139402626-527ac443-fe31-4054-ad42-92517363f614.png)

## 2. Some tech is missed in your codes.

The authors say "we recall the practical guidelines of an efficient CNN architecture [20] and build a lightweight network structure for EGVSR." But to be honest, I can not find codes using depthwise/pointwise/shuffle operations.
I can not find PyTorch code about MatMul, either. Do you implement them on FPGA? Could you provide code in the PyTorch framework?
Thanks for your time! It will be greatly appreciated if you can provide any guidance.

**Comment by cloudlakecho** [comment]
How to improve inference speed? I guess I have to smartly choose in these:  
&nbsp;&nbsp; Input image feeding (to neural network) method
&nbsp;&nbsp; Pre trained neural network weight selection
&nbsp;&nbsp; Hyper parameter
Any suggestion welcomed. 

The reason is that I am also experiencing performance speed degradation. The inference speed is around 341 ms per image (4 frame per second) even with more powerful GPU with smaller input image. I am using `codes/main.py` code with image from Vid4 dataset. My command line for inference: 
` python ./codes/main.py --exp_dir ./experiments/test_egvsr_vid4 --mode test --opt test.yml  --model EGVSR`
 It looks like the image fed as the group of from 40 to 60 to the neural network in the code.  

This is my runtime environment:
&nbsp;&nbsp; Input image size: 180 x 144 (Gaussian4xLR in Vid4 dataset)
&nbsp;&nbsp; Enhencement: 4x 
&nbsp;&nbsp; GPU: Nvidia GeForce RTX 3080 Ti 
&nbsp;&nbsp; Neural network weight: EGVSR_iter420000.pth

As reference: 
This is from the research paper: 
30 frame for second with
&nbsp;&nbsp; Input image size: 960 x 540
&nbsp;&nbsp; Enhencement: 4x 
&nbsp;&nbsp; GPU: Nvidia GeForce GTX 2080Ti GPU

**Comment by JacoboJin** [comment]
Yes, I also met this issue when I try to recurrent this model. 
This is my runtime environment:
   Input image size: 960 x 540 (Gaussian4xLR in Vid4 dataset)
   Enhencement: 4x
   GPU: NVIDIA RTX™ A5000
   Neural network weight: EGVSR_iter420000.pth

Meanwhile, the result is as below:
![image](https://user-images.githubusercontent.com/51702594/180789306-02ec6ae6-dc4b-44ba-af91-32f451fee660.png)
